packages/web3torrent/src/clients/web3torrent-client.ts

```diff
@@ -1,62 +1,29 @@
 import React from 'react';
 import {WebTorrentAddInput, WebTorrentSeedInput, ExtendedTorrent} from '../library/types';
-import WebTorrentPaidStreamingClient from '../library/web3torrent-lib';
+import WebTorrentLibrary from '../library/web3torrent-lib';
 import {Status} from '../types';
 import {paymentChannelClient} from './payment-channel-client';
-import {defaultTrackers, INITIAL_BUDGET_AMOUNT, FUNDING_STRATEGY} from '../constants';
-import _ from 'lodash';
-import {track} from '../analytics';
 
-export const web3TorrentClient = new WebTorrentPaidStreamingClient({
-  paymentChannelClient: paymentChannelClient,
-  tracker: {announce: defaultTrackers}
-});
+export const web3TorrentClient = new WebTorrentLibrary({paymentChannelClient});
 
 export const Web3TorrentClientContext = React.createContext(web3TorrentClient);
 
-export const getTorrentPeers = infoHash => web3TorrentClient.peersList[infoHash];
-
-const doesBudgetExist = async (): Promise<boolean> => {
-  const budget = await web3TorrentClient.paymentChannelClient.getBudget();
-  return !!budget && !_.isEmpty(budget);
-};
-
-export const download: (
-  torrent: WebTorrentAddInput
-) => Promise<ExtendedTorrent> = async torrentData => {
+export async function download(torrentData: WebTorrentAddInput): Promise<ExtendedTorrent> {
   await web3TorrentClient.enable();
-  if (FUNDING_STRATEGY !== 'Direct' && !(await doesBudgetExist())) {
-    await web3TorrentClient.paymentChannelClient.createBudget(INITIAL_BUDGET_AMOUNT);
-  }
 
-  return new Promise((resolve, reject) =>
+  return new Promise(resolve =>
     web3TorrentClient.add(torrentData, (torrent: any) =>
       resolve({...torrent, status: Status.Connecting})
     )
   );
-};
+}
 
-export const upload: (input: WebTorrentSeedInput) => Promise<ExtendedTorrent> = async input => {
+export async function upload(input: WebTorrentSeedInput): Promise<ExtendedTorrent> {
   await web3TorrentClient.enable();
-  // TODO: This only checks if a budget exists, not if we have enough funds in it
-  if (FUNDING_STRATEGY !== 'Direct' && !(await doesBudgetExist())) {
-    await web3TorrentClient.paymentChannelClient.createBudget(INITIAL_BUDGET_AMOUNT);
-  }
 
-  return new Promise((resolve, reject) =>
-    web3TorrentClient.seed(input, {...torrentNamer(input)}, (torrent: any) => {
-      resolve({
-        ...torrent,
-        status: Status.Seeding,
-        originalSeed: true
-      });
-    })
+  return new Promise(resolve =>
+    web3TorrentClient.seed(input, (torrent: any) =>
```

Comment:
Is there a reason to have callback style functions in `web3torrent-lib`? Perhaps the original intention was to match the signature of the overridden webtorrent functions. But do we think that's important?

At the moment, simplicity and clarity is likely more important than matching method signatures between webtorrent and web3torrent.

Reply:
I decided to just match the function signature of the original function. Maybe we could refactor those functions in another PR?